### PR TITLE
Split the test data downloading and test cases

### LINKED ISSUE #419: [Refactor] GitHub actions caching
Datasets are downloaded every time a test runs, this step is unnecessary and takes long time (>20 mins) to process. The aim is to remove all data download code from `test` folder and cache the data, so data can be called from cache each time running the test.

### PR DESCRIPTION
**Pull request by wang-hlin**
Fixes #419.

### Description
Split the code for downloading into `tests/test_data.py`. All downloaded data will be stored in the `test/test_data` directory. Some files will still be generated in the `test_data` folder while running `pytest`, as a result of data preprocessing.

Use `cfg.DATASET.DOWNLOAD` to control the download process in the `get_cifar` function located in `kale/loaddata/image_access.py`.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] In-line docstrings updated.
- [ ] [Source for documentation at `docs`](https://github.com/pykale/pykale/tree/main/docs/source) manually updated for new API.

### CODE REVIEW

**Review by xianyuanliu — CHANGES_REQUESTED** [review]
@wenruifan Please conduct a review.

**Review by wenruifan — CHANGES_REQUESTED** [review]
Good work with just some minor places to change.

**Review by wenruifan — APPROVED** [review]
Nice work.

File: tests/test_data.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+def download_path():
+    path = Path("tests") / "test_data"
+    path.mkdir(parents=True, exist_ok=True)
+    return str(path)
+
+
+path_test_data = download_path()
+
+
+# tests/conftest.py
```
Comment (xianyuanliu) [inline tests/test_data.py:21]: The comments are not clear. Please use a consistent format throughout, e.g. 'Downloading gait gallery data for tests/conftest.py test'. Please change all the comments below.

File: tests/test_data.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+url = "https://github.com/pykale/data/raw/main/multiomics/TCGA_BRCA.zip"
+dataset_root = str(Path(path_test_data) / "multiomics/trainer/multi_class/")
+download_file_by_url(url=url, output_directory=dataset_root, output_file_name="multi_class.zip", file_format="zip")
+
+# tests/prepdata/test_image_transform.py
+# Downloading CMR Dataset
+url = "https://github.com/pykale/data/raw/main/images/ShefPAH-179/SA_64x64_v2.0.zip"
+download_file_by_url(url, path_test_data, "SA_64x64.zip", "zip")
+
+office_path = str(Path(path_test_data) / "office")
```
Comment (xianyuanliu) [inline tests/test_data.py:64]: Is this still 'downloading CMR dataset'?

File: tests/test_data.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+
+datasets.MNIST(path_test_data, train=True, transform=get_transform("mnistm"), download=True)
```
Comment (xianyuanliu) [inline tests/test_data.py:67]: Is this still 'downloading CMR dataset'?

File: tests/test_data.py (hunk 2 above)
Comment (xianyuanliu) [reply tests/test_data.py:64]: Please put this line above the office data downloading.

**Review by xianyuanliu — CHANGES_REQUESTED** [review]

**Review by xianyuanliu — APPROVED** [review]
Well done. Please proceed with implementing data caching for downloads in GitHub Actions.

**Pull request merged**